File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/lesson/model/LessonCreateRequestDTO.java
package kh.farrukh.sortmapperexamplejava.lesson.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record LessonCreateRequestDTO(
    @NotBlank
    @Size(min = 2, max = 50)
    String title,
    @Size(min = 2, max = 200)
    String description,
    @NotNull
    Long courseId,
    Long helperTeacherId
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/lesson/model/LessonCreateRequestDTO.kt
package kh.farrukh.sortmapperexample.lesson.model

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class LessonCreateRequestDTO(
    @field:[NotBlank Size(min = 2, max = 50)]
    val title: String,
    @field:Size(min = 2, max = 200)
    val description: String?,
    val courseId: Long,
    val helperTeacherId: Long?
)
